config/300-kafkasource.yaml

```diff
@@ -0,0 +1,399 @@
+                    description: Kerberos Authentication method to interact with Kafka.
+                    type: object
+                    properties:
+                      kerberosRealm:
+                        description: The name of the Kerberos Realm.
+                        type: string
+                      kerberosServiceName:
+                        description: The primary name of the Kafka server configured.
+                        type: string
+                      username:
+                        description: Kerberos Username.
+                        type: string
+                      password:
+                        description: Kerberos Password.
+                        type: object
+                        properties:
+                          value:
+                            description: Plain text password.
+                            type: string
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the password.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        oneOf:
+                        - required: [value]
+                        - required: [valueFromSecret]
+                      kerberosConfig:
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        required:
+                        - valueFromSecret
+                      kerberosKeytab:
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        required:
+                        - valueFromSecret
+                  sslAuth:
```

Comment:
Better call this `tls`.
It might be authentication (if key is provided) or just SSL encryption (if only ca and crt are provided)